testapp/build.gradle

```diff
@@ -0,0 +1,69 @@
+plugins {
+    id 'com.android.application'
+}
+
+android {
+    namespace 'pl.brightinventions.slf4android.testapp'
+    compileSdk 34
+
+    defaultConfig {
+        applicationId "pl.brightinventions.slf4android.testapp"
+        minSdk 19
+        targetSdk 34
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    signingConfigs {
+        release {
+            storeFile file('testkeystore.jks')
+            storePassword 'teststore'
+            keyAlias 'key'
+            keyPassword 'testkey'
+        }
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            signingConfig signingConfigs.release
+        }
+    }
+
+    java {
+        toolchain {
+            languageVersion = JavaLanguageVersion.of(17)
```

Comment:
Isn't that what isn't recommended https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/ ? (TBH I didn't analyse the blog post in detail)

Reply:
Hmm, IMO yes and no.

Jake listed a few reasons not to use the toolchain but most of them seem to based on an assumption that the toolchain is used mainly as a replacement for `sourceCompatibility` and `targetCompatibility`.

My understanding of what he says is that if you purposefully use an old JDK version in the toolchain to set the compatibility with JDK 8, then you lose all the benefits of using the newer JDK.

However, if you set the compatibility levels separately from the toolchain (like I did in this PR), then most of his arguments don't seem to be valid or that important anymore.

I've just tested my hypothesis by adding `withJavadocJar()` here:

```groovy
publishing {
    singleVariant('release') {
        withSourcesJar()
        withJavadocJar() // <-- here
    }
}
```

and publishing it to Maven Local with ` ./gradlew :core:publishToMavenLocal`

then I unzipped and opened the generated Javadoc JAR, and I see it **does** contain a search bar:

![image](https://github.com/bright/slf4android/assets/5508894/6759c4b9-a26a-4776-9fac-6e7afe64db60)

So, assuming Jake was right about the search bar being a new feature of JDK 9, it clearly indicates the newer JDK is used to build the Javadoc, even though we keep the compatibility with JDK 8.

The toolchain setting is not so important to us and we could remove it without breaking anything as long as we keep the correct JDK in the build environment, but IMO that setting has at least three advantages:
- it's convenient 
- it gives the contributors a brief idea about the JDK version that is guaranteed to build the app/library without errors 
- it automatically downloads that version if they don't have it

If it wasn't for the Android Gradle Plugin's minimum requirement that is now JDK 17, I bet 99% of Android developers would stick to JDK 8 or JDK 11, which were the previously required versions (1-2 years ago) and were also distributed as the Android Studio's built-in JDKs.

Btw we can bump our toolchain version from 17 to 22 at any time and it should still work without any issues. I haven't done it in this PR only because this change was completely out of scope and I didn't want to waste any more time on such cosmetics.

Reply:
Also, since there is no good tool for managing multiple JDK versions for Windows that I'm aware of (e.g. I couldn't easily make SDKMAN work on my machine with Git Bash), this toolchain setting was my savior when I had to switch between the environments (e.g. from Ubuntu to macOS to Windows)